### ISSUE TITLE
Bluetooth necessary?

### ISSUE THREAD

**Issue by OneBobone** [bmork/defogger#1]
Impressive work!
I just got my rain-electrocuted device back up after several measures and a factory reset to FW 1.0.
The optics are a bit foggy, but hopefully that will go away.

My reflection is that it was not all that bad to use the dlink Bluetooth process to set up the device. I can still avoid the upgrade, etc...

Is the whole bluetooth set-up that you suggest avoidable if you disregard from this goal:  "configuration of network and admin password via Bluetooth LE, without registering with D-Link or using the mydlink app at all".
Perhaps it is necessary for the initial communication?

**Comment by bmork** [comment]
No, you don't have to do the Bluetooth part if you are happy with the dlink app. But Bluetooth is the only method I've looked into for running arbitrary commands, except using the serial console. One of those methods is necessarfy for the initial 2.x FW configuration, of you want to enable telnet, http and/or rtsp.

I don't haveany experience with the 1.0 FW, but I guess you then have access to the full http API by default?  That is probably sufficient for normal use.  You won't be able to start the telnet server or to install unsigned firmware.  But then again:  You don't need to do that :-)

I assume there are holes in the http API too, so that's probably also a way in.  But I haven't looked at it yet.  Started writing an Android app instead, to make the Bluetooth stuff a bit more accessible to those without a Linux laptop.  But I don't like Java, and I don't know the Android APIs very well, so I wouldn't hold my breath waiting for this work to finish....

**Comment by mcus** [comment]
An Android app would be so awesome (setting up and jailbreaking the camera ”on the go” between networks).
Thank you so much for taking the time to do this sir! :)

 I’m not too confident in myself to pull of what you show in the examples, so a ”simplified” step by step via an Android app would be so great.

Again, Thank you!

**Comment by OneBobone** [comment]
+1 
I would straight away borrow an Android to jail break a new camera (that I need)!

An easier alternative to writing a full Android app would be to port your solution to Cygwin on Windows?

**Comment by bmork** [comment]
Don't know if the Windows idea will work, and I have no way to test that. Cygwin is great, but access to hardware like the Bluetooth adapter isn't necessarily supported? Maybe someonw with Windows can try

**Comment by OneBobone** [comment]
I could maybe give it a try, but my Unix skills are extremely limited.
I have compiled and installed on Cygwin, but would need a possibly scaled down scope and stepwise instruction (necessary libraries to include, write this, do that).
Scaled down scope to start testing basics.😁

**Comment by bmork** [comment]
Please try.

This is a little premature, but if anyone is brave then they could also test out this Android app:
http://www.mork.no/~bjorn/defogger.apk

It's by no means a finished product, which will be obvious the instant you start it. I could claim that the UI is designed this way to make the pre-alpha status clear, but the fact is that I haven't bothered looking at the UI design at all yet. I know som parts is falling off the screen, for example...

Some features I would like to have is still missing:
 - network configuration
 - backup of camera flash
 - semi-automated firmware "fix"

But the most important features for initial access are working.  You can connect, authenticate and run arbitrary commands on the camera.  With absolutely no failsafe, just so I have said that.  And I have added the same default commands I had in the pyhton script.

This isn't really tested though.  I have only tried it on a single Android device (a Sony with Android 9), and the only camera target I have tested with is my already mydlink-free one.

I've pushed the source code to this branch: https://github.com/bmork/defogger/tree/devel-android-wip  in case anyone thinks an unverified apk from an unknown source is scary (you should!)

**Comment by OneBobone** [comment]
I am sorry, I have not had the possibility to try anything yet.
Actually, I managed to get hold of a few more with FW 1.01, so I am quite fine.

**Comment by bmork** [comment]
No problem.  I had to pause development due to lack of time.  Just pushed the latest I have to the "devel-android-wip" branch, and updated the binary apk to match the current devel HEAD.  Still in no way finished, but I believe it is now capable of initial configuring and "rooting" the camera.  Maybe even without crashing.

The UI still sucks of course, but at least i now supports reading the QR code so it's much easier to test again and again and again :-)

**Comment by dwildbore** [comment]
how did you find the firmware link
